### ISSUE TITLE
LogFC sign in sigGenes.results.txt

### ISSUE THREAD

**Issue by ewilkie** [hartleys/JunctionSeq#24]
Hi,

I have a concern regarding the sign of the logFC in the sigGenes.results.txt file. I'm using the logFC to split those junctions into groups depending whether they are upregulated in KO or downregulated in KO. For certain junctions based on the expr_KO and expr_WT values I expect the sign to be the opposite. Can you please explain what is causing this behaviour? 

In the attached file I would expect event 3 and 4 to have a positive logFC value.
[issue.csv.zip](https://github.com/hartleys/JunctionSeq/files/1241566/issue.csv.zip)

Thanks in advance for your time :)

**Comment by hartleys** [comment]
JunctionSeq is designed to detect junctions and exons for which the expression does not match the expression of the gene as a whole. It does not simply detect which junctions have different coverage levels in cases vs controls, since this would be predominately composed of junctions belonging to genes that are differentially expressed overall.

Thus, the "fold change" is actually a fold change relative to the gene-level expression. 

So, to provide an extreme example: if the overall gene is 4x upregulated in WT, and the junction shows ***NO DIFFERENCE***, then the junction will be labelled as differentially used, and the junction fold change will be -4. This is even though expr_KO EQUALS expr_WT for this junction, since based on the gene as a whole KO is 4x bigger than expected relative to WT. (It's actually a little more complex than that, but that's the basic idea.)

If you look at the "sigGenes.expression.data.txt.gz" output file, you will see columns labelled "geneExpr_WT" and "geneExpr_KO", which will show the expression for the rest of the gene not counting the reads on each given splice junction. The "fold change" is calculated as:

    log2FC(WT/KO) = log2( ( expr_WT / exprGene_WT ) / ( expr_KO / exprGene_KO )  )

**Comment by ewilkie** [comment]
Thank you for the explanation. I appreciate that the junction should not be viewed out of context of the gene as a whole :)

Using the columns you mentioned I calculated the ratio of the junction expression vs gene expression per condition.

WT_ratio = ( expr_WT / exprGene_WT )
KO_ratio = ( expr_KO / exprGene_KO )

For event 3 and 4, the WT_ratio is smaller than the KO_ratio, despite higher expr_WT. 
[LogFC_example.csv.zip](https://github.com/hartleys/JunctionSeq/files/1244357/LogFC_example.csv.zip)

So in other words, there is less expression of the WT junction compared to the KO junction, in relation to the respective gene expression.

**Comment by hartleys** [comment]
Exactly. Does this all make sense?

**Comment by ewilkie** [comment]
So far so good. However I'm wondering what implications this has in scenarios where a large amount of nascent transcripts are present. Particularly those where only partial splicing has occurred. This could potentially skew the overall count of the gene and hence throw off the ratios. If the amount of nascent vs mature mRNA has been quantified in the same sample, this could be tested.
